src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -93,6 +94,12 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
             return;
         }
 
+        Pattern usernamePattern = Pattern.compile("[$&+,:;=\\\\?@#|/'<>^*()%!-]");
```

Comment:
> Username should not include special characters

Why not, are there bugs associated with these characters?

Reply:
Thanks for looking out for issues of this nature, could you be more specific as to the risks associated with individual characters?  I am reticent to limit these usernames without cause.

If there is an issue please follow our reporting process [1] also quoted below.

> ## Reporting a Vulnerability
>
>If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/) or directly via email to aws-security@amazon.com. Please do **not** create a public GitHub issue.

[1] https://github.com/opensearch-project/security/security/policy

Reply:
Added more details on the main thread https://github.com/opensearch-project/security/pull/2277#issuecomment-1333993589